libs/execution_context/ebpf_core.c

```diff
@@ -2206,15 +2206,18 @@ _ebpf_core_protocol_map_async_query(
     UNREFERENCED_PARAMETER(reply_length);
 
     ebpf_map_t* map = NULL;
+    uint32_t index = 0;
 
     ebpf_result_t result =
         EBPF_OBJECT_REFERENCE_BY_HANDLE(request->map_handle, EBPF_OBJECT_MAP, (ebpf_core_object_t**)&map);
     if (result != EBPF_SUCCESS) {
         goto Exit;
     }
 
+    index = request->index;
+
     // Return buffer already consumed by caller in previous notification.
-    result = ebpf_map_return_buffer(map, request->index, request->consumer_offset);
+    result = ebpf_map_return_buffer(map, index, request->consumer_offset);
```

Comment:
do we need a check in ebpfcore that index should be == 0 when map type is ringbuf?